### PR TITLE
feat: token secutiry info

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
image

### CODE REVIEW

File: src/components/CurrencyLogo/index.tsx [hunk 1]
```diff
@@ -14,6 +14,8 @@ import { getProxyTokenLogo } from 'utils/tokenInfo'
 const StyledNativeCurrencyLogo = styled.img<{ size: string }>`
   width: ${({ size }) => size};
   height: ${({ size }) => size};
+  min-width: ${({ size }) => size};
+  min-height: ${({ size }) => size};
```
Comment (namgold) [inline src/components/CurrencyLogo/index.tsx:18]: 🤔 để vậy nó có kéo dãn hình nhìn sida ko?

File: src/services/coingecko.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      query: ({ chainId, address }) => ({
+        url: `https://api.gopluslabs.io/api/v1/token_security/${chainId}?contract_addresses=${address}`,
+      }),
+      transformResponse: (data: any, _, arg) => data?.result?.[arg.address.toLowerCase()],
+    }),
+  }),
+})
+
+// todo tracking
+// todo (not for now) move basic chart api to this file
```
Comment (namgold) [inline src/services/coingecko.ts:51]:
```suggestion
// todo Danh (not for now) move basic chart api to this file
```

**Review by namgold — APPROVED** [review]

File: src/components/swapv2/TokenInfo/MarketInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  color: ${({ theme }) => theme.text};
+  font-size: 12px;
+  font-weight: 500;
+`
+export default function MarketInfo({ token }: { token: Token | undefined }) {
+  const { data: tokenInfo, loading } = useTokenInfo(token)
+  const [expand, setExpand] = useState(false)
+  const { chainId } = useActiveWeb3React()
+  const theme = useTheme()
+  const listData = [
```
Comment (XiaoYhun) [inline src/components/swapv2/TokenInfo/MarketInfo.tsx:45]: useMemo here? Its generate new list on every render

File: src/components/swapv2/TokenInfo/MarketInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          <InfoRowValue>{loading ? <Loader size="10px" /> : item.value}</InfoRowValue>
+        </RowBetween>
+      ))}
+
+      <RowBetween style={{ borderBottom: 'none', paddingBottom: 0 }}>
+        <InfoRowLabel>
+          <Trans>Contract Address</Trans>
+        </InfoRowLabel>
+
+        <Flex width="fit-content" alignItems={'center'} sx={{ gap: '4px' }}>
```
Comment (XiaoYhun) [inline src/components/swapv2/TokenInfo/MarketInfo.tsx:90]: We can replace all Flex components like this to use pre-defined Row or Column components? This way, we can avoid repeatedly adding alignItems='center' everywhere.
`<RowFit gap="4px>`

File: src/components/Collapse.tsx [hunk 5]
```diff
@@ -56,6 +56,9 @@ type Props = {
   onExpand?: () => void
   className?: string
   arrowComponent?: ReactNode
+  headerStyle?: CSSProperties
+  headerBorderRadius?: string
+  arrowStyle?: CSSProperties
 }
 
 export const CollapseItem: React.FC<Props> = ({
```
Comment (XiaoYhun) [inline src/components/Collapse.tsx:64]: I suggest add toggle animation for this component as we have good UI here but quite a bad visual.
Also I see the Header color of these toggle are different than design

Comment (nguyenhoaidanh) [reply src/components/Collapse.tsx:72]: nice catch

File: src/components/swapv2/TokenInfo/MarketInfo.tsx (hunk 4 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/TokenInfo/MarketInfo.tsx]: cool, I event don't know the exist of this component RowFit

File: src/components/CurrencyLogo/index.tsx (hunk 1 above)
Comment (nguyenhoaidanh) [reply src/components/CurrencyLogo/index.tsx:18]: kéo là kéo sao fen. chưa hiểu.
này thêm vô để fix 1 số case width của thằng cha bé quá nó mất token logo

**Comment by namgold** [comment]
build fail

File: src/components/CurrencyLogo/index.tsx (hunk 1 above)
Comment (namgold) [reply src/components/CurrencyLogo/index.tsx:18]: isn't width enogh?

Comment (namgold) [reply src/components/CurrencyLogo/index.tsx:18]: If you have tested this carefully, it's ok to resolve this comment

Comment (nguyenhoaidanh) [reply src/components/CurrencyLogo/index.tsx]: just tested again, it's ok to remove min-height

**Comment by nguyenhoaidanh** [comment]
build ok now @namgold

**Review by namgold — APPROVED** [review]

**Pull request merged**